x-pack/plugins/actions/README.md

```diff
@@ -27,12 +27,12 @@ Table of Contents
     - [Example](#example)
   - [RESTful API](#restful-api)
     - [`POST /api/action`: Create action](#post-apiaction-create-action)
-    - [`DELETE /api/action/{id}`: Delete action](#delete-apiactionid-delete-action)
-    - [`GET /api/action/_getAll`: Get all actions](#get-apiactiongetall-get-all-actions)
-    - [`GET /api/action/{id}`: Get action](#get-apiactionid-get-action)
-    - [`GET /api/action/types`: List action types](#get-apiactiontypes-list-action-types)
-    - [`PUT /api/action/{id}`: Update action](#put-apiactionid-update-action)
-    - [`POST /api/action/{id}/_execute`: Execute action](#post-apiactionidexecute-execute-action)
+    - [`DELETE /api/actions/{id}`: Delete action](#delete-apiactionid-delete-action)
+    - [`GET /api/actions`: Get all actions](#get-apiactiongetall-get-all-actions)
+    - [`GET /api/actions/{id}`: Get action](#get-apiactionid-get-action)
+    - [`GET /api/actions/list_action_types`: List action types](#get-apiactiontypes-list-action-types)
```

Comment:
This still has a problem - if a customer creates a preconfigured action with id `list_action_types`.  

I think we need to create a new path element - say `action` (but of course, that's not great), that we'd use for all the `{id}` CRUD operations.  Here's some URIs that I think will work (current ones from PR on left, suggestions on the right):

```
POST   /api/actions                     -> POST   /api/actions/action
GET    /api/actions/{id}                -> GET    /api/actions/action/{id}
PUT    /api/actions/{id}                -> PUT    /api/actions/action/{id}
DELETE /api/actions/{id}                -> DELETE /api/actions/action/{id}
POST   /api/actions/{id}/_execute       -> POST   /api/actions/action/{id}/_execute
GET    /api/actions                     -> GET    /api/actions
GET    /api/actions/list_action_types   -> GET    /api/actions/list_action_types
```

If we end up having "preconfigured alerts", we may end up with the same (or very similar) problem.  I wonder if we should look at all the URLs for alerts, actions, event log, (task manager?), and see if there is some common pattern we can apply to all of them.

Reply:
I think it makes sense and I didn't cover it in the current implementation. 
For alerts we can have `/api/alerts/alert` and it will be the same approach.  
As for the event logs it is a bit different, because we have logs a type specific `/api/event_log/{type}` - something like `/api/event_log/alerts` and `/api/event_log/actions`. But I'm not sure for 100%, if we want to have it only like this for GET operations or may need additionally the general approach like `/api/event_logs` to get all the logs. And what about CRUD operations for the event log - are we planning to expose the routes for this?

Reply:
> And what about CRUD operations for the event log - are we planning to expose the routes for this?

Realizing that in terms of http routes, there is only Read (Query) for event log :-)

And I think the event log query shape is likely to change.  We'll probably want to be able to send a list of SO's by type/id ; and we will probably want to filter some documents.  But TBD, and I don't think it should block getting alerting and action routes fixed up.  So let's ignore the event log for now ...